internal/cli/atlas/clusters/region_tier_autocomplete.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusters
+
+import (
+	"context"
+	"sort"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/validate"
+	"github.com/spf13/cobra"
+)
+
+type autoFunc func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective)
+
+type autoCompleteOpts struct {
+	cli.GlobalOpts
+	providers []*string
+	tier      string
+	store     store.CloudProviderRegionsLister
+}
+
+func (opts *autoCompleteOpts) autocompleteTier() autoFunc {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		opts.parseFlags(cmd)
+		if err := validate.Credentials(); err != nil {
+			cobra.CompErrorln("no credentials")
+			return []string{}, cobra.ShellCompDirectiveError
```

Comment:
in the [go wiki](https://github.com/golang/go/wiki) they prefer nil over empty slices [ref](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices)

have you checked if 
```suggestion
			return nil, cobra.ShellCompDirectiveError
```

produces unintended behaviours?

Reply:
I can check this out, I'm pretty sure I was following (copy/pasting) some examples from cobra give a sec to see if this works well with nil

Reply:
updated, I don't see any major issue